src/connection/mc_worker.erl

```diff
@@ -189,10 +189,11 @@ get_write_concern(_) -> undefined.
 %% @private
 %% Parses proplist to record
 form_state(Options) ->
-  Database = mc_utils:get_value(database, Options, <<"admin">>),
+    Database = mc_utils:get_value(database, Options, <<"admin">>),
+    AuthSource = mc_utils:get_value(auth_source, Options, <<"admin">>),
   RMode = mc_utils:get_value(r_mode, Options, master),
   WMode = mc_utils:get_value(w_mode, Options, unsafe),
-  #conn_state{database = Database, read_mode = RMode, write_mode = WMode}.
+  #conn_state{database = Database, auth_source = AuthSource, read_mode = RMode, write_mode = WMode}.
```

Comment:
I guess you should add `auth_source` to `#conn_state` also.

Reply:
Of course. I forgot to add mongo_protocol.hrl to the commit. My bad!